internal/datastore/mysql/revisions.go

```diff
@@ -0,0 +1,178 @@
+	// TODO (@vroldanbet) dupe from postgres datastore - need to refactor
+	ctx, span := tracer.Start(ctx, "CheckRevision")
+	defer span.End()
+
+	revisionTx := transactionFromRevision(revision)
+
+	freshEnough, future, err := mds.checkValidTransaction(ctx, revisionTx)
+	if err != nil {
+		return fmt.Errorf(errCheckRevision, err)
+	}
+
+	if !freshEnough {
+		return datastore.NewInvalidRevisionErr(revision, datastore.RevisionStale)
+	}
+	if future {
+		return datastore.NewInvalidRevisionErr(revision, datastore.RevisionInFuture)
+	}
+
+	return nil
+}
+
+func (mds *Datastore) loadRevision(ctx context.Context) (uint64, error) {
+	// TODO (@vroldanbet) dupe from postgres datastore - need to refactor
+	// slightly changed to support no revisions at all, needed for runtime seeding of first transaction
+	ctx, span := tracer.Start(ctx, "loadRevision")
+	defer span.End()
+
+	query, args, err := mds.GetLastRevision.ToSql()
+	if err != nil {
+		return 0, fmt.Errorf(errRevision, err)
+	}
+
+	var revision sql.NullInt64
+	err = mds.db.QueryRowContext(datastore.SeparateContextWithTracing(ctx), query, args...).Scan(&revision)
+	if err != nil {
+		if errors.Is(err, sql.ErrNoRows) {
+			return 0, nil
+		}
+		return 0, fmt.Errorf(errRevision, err)
+	}
+	if !revision.Valid {
+		// there are no rows in the relation tuple transaction table
+		return 0, nil
+	}
+
+	return uint64(revision.Int64), nil
+}
+
+func (mds *Datastore) checkValidTransaction(ctx context.Context, revisionTx uint64) (bool, bool, error) {
+	ctx, span := tracer.Start(ctx, "checkValidTransaction")
+	defer span.End()
+
+	var freshEnough, future sql.NullBool
+
+	err := mds.db.QueryRowContext(
+		datastore.SeparateContextWithTracing(ctx), mds.validTransactionQuery, revisionTx, revisionTx,
+	).Scan(&freshEnough, &future)
+	if err != nil {
+		return false, false, fmt.Errorf(errCheckRevision, err)
+	}
+
+	span.AddEvent("DB returned validTransaction checks")
+
+	return freshEnough.Bool, future.Bool, nil
+}
+
+func (mds *Datastore) createNewTransaction(ctx context.Context, tx *sql.Tx) (newTxnID uint64, err error) {
+	ctx, span := tracer.Start(ctx, "createNewTransaction")
+	defer span.End()
+
+	createQuery := mds.createTxn
+	if err != nil {
+		return 0, fmt.Errorf("createNewTransaction: %w", err)
+	}
+
+	result, err := tx.ExecContext(ctx, createQuery)
+	if err != nil {
+		return 0, fmt.Errorf("createNewTransaction: %w", err)
+	}
+
+	lastInsertID, err := result.LastInsertId()
+	if err != nil {
+		return 0, fmt.Errorf("createNewTransaction: failed to get last inserted id: %w", err)
+	}
+
+	return uint64(lastInsertID), nil
+}
+
+// TODO (@vroldanbet) dupe from postgres datastore - need to refactor
+func revisionFromTransaction(txID uint64) datastore.Revision {
+	return decimal.NewFromInt(int64(txID))
+}
+
+// TODO (@vroldanbet) dupe from postgres datastore - need to refactor
+func transactionFromRevision(revision datastore.Revision) uint64 {
+	return uint64(revision.IntPart())
+}
```

Comment:
```suggestion
func revisionFromTransaction(txID uint64) datastore.Revision {
	return decimal.NewFromBigInt(new(big.Int).SetUint64(txID), 0)
}

func transactionFromRevision(revision datastore.Revision) uint64 {
	return revision.BigInt().Uint64()
}
```

I think something like this will work, but I haven't tested it. Note that this does differ from the postgres implementation; postgres's `BIGSERIAL` is a signed int, but the revision field in the mysql schema is `BIGINT UNSIGNED`.

Reply:
Changed and added tests in https://github.com/authzed/spicedb/pull/582/commits/b93e5c083e7794e241c01f2a5426cf61bc112c6d